From b6c32f0bb490a9cd1e1f65e995e45c72da2394f6 Mon Sep 17 00:00:00 2001
From: Sam Mortimer <sam@mortimer.me.uk>
Date: Tue, 14 Aug 2018 19:33:50 -0700
Subject: [PATCH 4/8] LineageSettingsProvider: Don't access system settings
 during startup

*) This is not possible since:
https://android.googlesource.com/platform/frameworks/base/+/235272a2165b694ca1d67e901aa2ef2ad86c35bc

*) Unblock pie sdk bringup by commenting out the (only) system
   settings access that happens during LineageSettingsProvider
   startup.

*) We should later determine a proper fix prior to shipping lineage suw.
   eg perhaps suw should check Settings.Global.DEVICE_PROVISIONED instead.

Change-Id: I185648ca25995fae80362642d297573795d04bfa
---
 .../org/lineageos/lineagesettings/LineageDatabaseHelper.java  | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
index f245dd9..9eaf7f8 100644
--- a/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
+++ b/packages/LineageSettingsProvider/src/org/lineageos/lineagesettings/LineageDatabaseHelper.java
@@ -420,9 +420,13 @@ public class LineageDatabaseHelper extends SQLiteOpenHelper{
                     LineageSettings.Secure.PROTECTED_COMPONENT_MANAGERS,
                     R.string.def_protected_component_managers);
 
+            /* TODO: Determine a proper fix prior to shipping lineage suw.
+             *       eg perhaps suw should check Settings.Global.DEVICE_PROVISIONED
+             *       instead.
             final String provisionedFlag = Settings.Global.getString(mContext.getContentResolver(),
                     Settings.Global.DEVICE_PROVISIONED);
             loadSetting(stmt, LineageSettings.Secure.LINEAGE_SETUP_WIZARD_COMPLETED, provisionedFlag);
+            */
         } finally {
             if (stmt != null) stmt.close();
         }
-- 
2.17.1

